tools/stack_decode.py

```diff
@@ -70,38 +51,41 @@ def decode_stacktrace_log(input_source):
 # to resolve, one per line.
 #
 # Returns list of result lines
-def run_addr2line(obj_file, piped_input):
-  addr2line = subprocess.Popen(["addr2line", "-Cpisfe", obj_file],
-                               stdin=subprocess.PIPE,
+def run_addr2line(obj_file, addr_to_resolve):
+  addr2line = subprocess.Popen(["addr2line", "-Cpie", obj_file, addr_to_resolve],
                                stdout=subprocess.PIPE)
-  output_stdout, _ = addr2line.communicate(piped_input)
-  return output_stdout.split("\n")
+  output_stdout, _ = addr2line.communicate()
+  return output_stdout
 
 
-# Output one stacktrace after passing it through addr2line with appropriate
-# options
-def output_stacktrace(thread_id, traceinfo):
-  output_lines = []
-  for address_list in traceinfo.obj_list:
-    piped_input = ""
-    obj_name = address_list.obj_file
-    for stack_addr in address_list.addresses:
-      piped_input += (stack_addr + "\n")
-    output_lines += run_addr2line(obj_name, piped_input)
+# Because of how bazel compiles, addr2line reports file names that begin with
+# "/proc/self/cwd/" and sometimes even "/proc/self/cwd/./". This isn't particularly
+# useful information, so trim it out and make a perfectly useful relative path.
+def trim_proc_cwd(file_and_line_number):
+  # Try the longer trim first, it will leave the string untouched if it doesn't
+  # match.
+  file_and_line_number = trim_from_start(file_and_line_number, "/proc/self/cwd/./")
+  file_and_line_number = trim_from_start(file_and_line_number, "/proc/self/cwd/")
+  return file_and_line_number
+
+
+def trim_from_start(string, trim):
```

Comment:
Nit: prefer not using `string` here, since it can be confused with https://docs.python.org/3/library/string.html